osu.Game/Screens/OnlinePlay/DrawableRoomPlaylistItem.cs

```diff
@@ -157,7 +163,13 @@ protected override void LoadComplete()
 
                         if (Item.Beatmap.Value == null)
                         {
-                            var foundBeatmap = await beatmapLookupCache.GetBeatmapAsync(Item.BeatmapID).ConfigureAwait(false);
+                            IBeatmapInfo foundBeatmap;
+
+                            if (multiplayerClient != null)
+                                foundBeatmap = await multiplayerClient.GetAPIBeatmap(Item.BeatmapID).ConfigureAwait(false);
+                            else
+                                foundBeatmap = await beatmapLookupCache.GetBeatmapAsync(Item.BeatmapID).ConfigureAwait(false);
```

Comment:
This `MultiplayerClient` method eventually needs to be removed..

I've done this for the time being so that `DrawablePlaylistItem` queries via the same method as `MatchSubScreen`, in order for the panels to arbitrarily replace the beatmap with an unexpected model in tests. Keep in mind that `TestMultiplayerClient` overrides `GetAPIBeatmap()` to do a more specialised lookup.

Reply:
I've added an inline comment mentioning this as it's a bit of a wtf when reading the code.